### Back to the system of equations

In the last lesson, we saw how we can use vector multiplication to predict target variables.

For example, let's assume we would like to predict sales where $ad\_spend = 1000 $ and $ t\_shirt\_price = 10$.  Let's also assume that the coefficients for ad spend and t shirt price are $.35$ and $-12$ respectively.  Then we can let $a$ be a vector that represents our feature variables, and $x$ represent our coefficients such that: 

$a = \begin{pmatrix}
    1000 \\
    10 \\
\end{pmatrix}$

$x = \begin{pmatrix}
    .35 \\
    -12 \\
\end{pmatrix}$

And we calculate this in numpy with the following:

In [4]:
import numpy as np
a = np.array([1000, 10])
x = np.array([.35, -12])

a.dot(x)

230.0

More generally, $a * x = a_{i} x_{i} + a_{i+1} x_{i + 1} ... + a_{n} x_{n} $ where $a \in \mathbb{R}^n $ and $x \in \mathbb{R}^n $.

### Moving beyond one equation

Now in most scenarios, we would not want to make just one prediction, but multiple.  Let's set our feature variables as the following matrix A, where each row represents a different set of feature variables:

$A = \begin{pmatrix}
    1000 & 10 \\
    2500 & 12 \\
    600 & 8 \\
    3000 & 14 \\ 
\end{pmatrix}$

Now with regression, we still would just have two coefficients one for ad spend and one for price.  We multipy these two numbers by each row to get the predicted number of T-shirt sales.  So $x$ is the same vector as before:

$x = \begin{pmatrix}
    .35 \\
    -12 \\
\end{pmatrix}$

Now let's try to multiply $A*x$.

In [8]:
x = np.array([.35, -12])
A = np.array([
    [1000, 10],
    [2500, 12],
    [600, 8],
    [3000, 14],
])

In [12]:
A.dot(x)

array([230., 731., 114., 882.])

It makes sense that the return value is a vector as it is saying the following:

$$1000*.35 + 10*-12 = 230 $$

$$2500*.35 + 12*-12=  731 $$

$$600*.35 + 8*-12=  114$$

$$3000*.35 + 14*-12 =  882$$

### Performing the calculation

Where we have a matrix A and a vector $x$ where A and x equal the following: 

$A = \begin{pmatrix}
    1000 & 10 \\
    2500 & 12 \\
    600 & 8 \\
    3000 & 14 \\ 
\end{pmatrix}$

$x = \begin{pmatrix}
    .35 \\
    -12 \\
\end{pmatrix}$

We can think of $A \cdot x$ as the following.

$ .35 * \begin{pmatrix}
    1000  \\
    2500  \\
    600  \\
    3000  \\ 
\end{pmatrix} + -12 * \begin{pmatrix}
    10  \\
    12  \\
    8  \\
    14  \\ 
\end{pmatrix} $

In terms of how to think about this, we can think of each column in A as a separate vector.

$x = \begin{pmatrix}
    1000  \\
    2500  \\
    600  \\
    3000  \\ 
\end{pmatrix}$
    
$y = \begin{pmatrix}
    10  \\
    12  \\
    8  \\
    14  \\ 
\end{pmatrix}$

In [14]:
A[:, 0]

array([1000, 2500,  600, 3000])

We can rewrite this equation as the following:

$ax = b $ where

$a = \begin{pmatrix}
    800 \\
    13 \\
\end{pmatrix}$

$x = \begin{pmatrix}
    x_1 \\
    x_2 \\
\end{pmatrix}$

and $b$ = 330

And if we solve for values of $x_1$ and $x_2$ we can also use the dot product to predict new values.  For example, let's assume $x_1$ is .25 and $x_1 = -10$.  